src/runtime/exceptions.cs

```diff
@@ -85,6 +85,25 @@ public new static IntPtr tp_str(IntPtr ob)
             }
             return Runtime.PyUnicode_FromString(message);
         }
+        
+        static void ClearOffsetHelper(IntPtr ob, int offset)
+        {
+            var field = Marshal.ReadIntPtr(ob, offset);
+            Runtime.Py_CLEAR(ref field);
+        }
```

Comment:
In C# 9 it should be possible to do a `ref` return. E.g. we could have `ref T Util.Ref<T>(IntPtr ob, int offest) where T: unmanaged`, then pass the reference to `Py_CLEAR`

Reply:
BTW, the way you are doing right now, you are not nulling the field in the original object.

Reply:
As it stands, I think this is just equivalent to `Runtime.XDecref(field)`, only the local variable `field` is set to `null`.

Reply:
Oh right. That's *not* the cpython macro Py_CLEAR